packages/fxa-auth-server/lib/payments/stripe.js

```diff
@@ -745,28 +746,20 @@ class StripeHelper {
         sub.latest_invoice &&
         sub.status === 'past_due' &&
         typeof sub.latest_invoice !== 'string' &&
-        sub.collection_method === 'charge_automatically' &&
-        typeof sub.latest_invoice.charge === 'string'
+        sub.collection_method === 'charge_automatically'
       ) {
-        const charge = await this.stripe.charges.retrieve(
-          sub.latest_invoice.charge
-        );
-        failure_code = charge.failure_code;
-        failure_message = charge.failure_message;
+        let charge = sub.latest_invoice.charge;
+        if (typeof sub.latest_invoice.charge === 'string') {
```

Comment:
I updated this logic because previously, if the charge happened to already be expanded, the failure information would not be added.